### ISSUE TITLE
XPath and XQuery 3.1?

### ISSUE THREAD

**Issue by joewiz** [benibela/xidel#37]
Hi Benito,

A [stack overflow question](https://stackoverflow.com/questions/58785466/how-to-join-two-extracted-values-with-xidel) led me to install xidel (via [homebrew](https://formulae.brew.sh/formula/xidel)) and take it for a spin. It's really nice work - I'm enjoying running XQuery very rapidly on remote URLs via the command line. 

I'm wondering if you have any plans to support XPath and XQuery 3.1 functions? I found myself reaching for `fn:sort`, for example. If there's already partial support, do you have a list of implemented/unimplemented functions? Are there any other caveats about the language level supported in xidel? Have you have run xqts on the XQuery engine?

Thanks!
Joe

**Comment by Reino17** [comment]
The latest [development snapshots](https://sourceforge.net/projects/videlibri/files/Xidel/Xidel%20development/) already contain some XQuery 3.1 functions. If I'm correct then the next stable release will have full XQuery 3.1 support.

[Xidel / Internet Tools function list](http://www.benibela.de/documentation/internettools/xpath-functions.html)

**Comment by benibela** [comment]
The 3.1 functions are almost all there, but the data structures are not. Biggest issue is that my maps have strings as keys, but 3.1 maps can have keys of any type. I need to completely rewrite the maps

Here are the old xqts 3.0 results of the last release: http://benibela.de/documentation/internettools/xqts.html

Here is the current state for 3.1: [xquery31.html.gz](https://github.com/benibela/xidel/files/3840310/xquery31.html.gz)

**Comment by benibela** [comment]
Now I have implemented all functions 

Although there are still failing tests [xqts.zip](https://github.com/benibela/xidel/files/5270087/xqts.zip)

I used your xml-to-json/json-to-xml functions, you know how they are. Some places still have jsoniq/string maps.   libicu does not support `alternate=blanked` for UCA collations. The regex library only matches assigned codepoints. The JSON parser does not do fallback escapes
